RocketChat/Rocket.Chat.ReactNative
# [NEW] Support Google OAuth from external browser

## Proposed changes

Adds support to Google OAuth from external browsers.

## Issue(s)	

Depends on https://github.com/RocketChat/Rocket.Chat/pull/22014

## How to test or reproduce

- Configure Google OAuth on local server using https://github.com/RocketChat/Rocket.Chat/pull/22014 branch
- Tapping Google should open browser and request login
- Callback should redirect back to the app as a deep link

## Screenshots

https://user-images.githubusercontent.com/804994/118133705-84a9e780-b3d7-11eb-9690-1cea398ab8ac.mp4



## Types of changes




## Checklist



## Further comments
